tests/examples/testWorkflowExamples.cpp

```diff
@@ -0,0 +1,104 @@
+    std::vector<double> timestampsBuffer(bufferSize);
+
+    std::vector<Types::ChannelVector> mockRecordingArrays =
+        getMockChannelArrays();
+    std::vector<std::vector<float>> mockData =
+        getMockData2D(numSamples, numChannels);
+    std::vector<double> mockTimestamps = getMockTimestamps(numSamples);
+
+    std::string path = getTestFilePath("exampleRecording.nwb");
+    // [example_workflow_io_snippet]
+    std::shared_ptr<BaseIO> io = createIO("HDF5", path);
+    // [example_workflow_io_snippet]
+
+    // [example_workflow_recording_containers_snippet]
+    std::unique_ptr<NWB::RecordingContainers> recordingContainers =
+        std::make_unique<NWB::RecordingContainers>();
+    // [example_workflow_recording_containers_snippet]
+
+    // [example_workflow_nwbfile_snippet]
+    std::unique_ptr<NWB::NWBFile> nwbfile =
+        std::make_unique<NWB::NWBFile>(generateUuid(), io);
+    nwbfile->initialize();
+    // [example_workflow_nwbfile_snippet]
+
+    // [example_workflow_datasets_snippet]
+    nwbfile->createElectricalSeries(
+        mockRecordingArrays, BaseDataType::I16, recordingContainers.get());
+    // [example_workflow_datasets_snippet]
+
+    // [example_workflow_start_snippet]
+    io->startRecording();
+    // [example_workflow_start_snippet]
+
+    // write data during the recording
+    bool isRecording = true;
+    while (isRecording) {
+      // write data to the file for each channel
+      for (SizeType i = 0; i < mockRecordingArrays.size(); ++i) {
+        const auto& channelVector = mockRecordingArrays[i];
+        for (const auto& channel : channelVector) {
+          // copy data into buffer
+          std::copy(mockData[channel.globalIndex].begin() + samplesRecorded,
+                    mockData[channel.globalIndex].begin() + samplesRecorded
+                        + bufferSize,
+                    dataBuffer.begin());
+          std::copy(mockTimestamps.begin() + samplesRecorded,
+                    mockTimestamps.begin() + samplesRecorded + bufferSize,
+                    timestampsBuffer.begin());
+
+          // write timeseries data
+          std::vector<SizeType> positionOffset = {samplesRecorded,
+                                                  channel.localIndex};
+          std::vector<SizeType> dataShape = {dataBuffer.size(), 1};
+          std::unique_ptr<int16_t[]> intBuffer = transformToInt16(
+              dataBuffer.size(), channel.getBitVolts(), dataBuffer.data());
+
+          // [example_workflow_write_snippet]
+          recordingContainers->writeTimeseriesData(i,
+                                                   channel,
+                                                   dataShape,
+                                                   positionOffset,
+                                                   intBuffer.get(),
+                                                   timestampsBuffer.data());
+          // [example_workflow_write_snippet]
```

Comment:
Should we add an optional call to io->flush() so we can add a note in the docs that calling flush ensures that is moved to disk?